File: ABB-setup/dv-processing-rel_1.7/docs/source/assets/code_samples/c++/writing_data/sample4.cpp
#include <dv-processing/io/mono_camera_writer.hpp>

int main() {
	// Sample VGA resolution, same as the DVXplorer camera
	const cv::Size resolution(640, 480);

	// Frame only configuration
	const auto config = dv::io::MonoCameraWriter::FrameOnlyConfig("DVXplorer_sample", resolution);

	// Create the writer instance, it will only have a single frame output stream.
	dv::io::MonoCameraWriter writer("mono_writer_sample.aedat4", config);

	// Write 10 image frames
	for (int i = 0; i < 10; i++) {
		// Initialize a white image
		cv::Mat image(resolution, CV_8UC3, cv::Scalar(255, 255, 255));

		// Generate some monotonically increasing timestamp
		const int64_t timestamp = i * 1000;

		// Encapsulate the image in a frame that has a timestamp, this does not copy the pixel data
		dv::Frame frame(timestamp, image);

		// Write the frame
		writer.writeFrame(frame);
	}

	return 0;
}

File: source_code/ABB-setup/dv-processing-rel_1.7/docs/source/assets/code_samples/python/writing_data/sample4.py
import dv_processing as dv
import numpy as np

# Frame only configuration
config = dv.io.MonoCameraWriter.FrameOnlyConfig("DVXplorer_sample", (640, 480))

# Create the writer instance, it will only have a single frame output stream
writer = dv.io.MonoCameraWriter("mono_writer_sample.aedat4", config)

# Write 10 image frames
for i in range(10):
    # Initialize a white image
    image = np.full((480, 640, 3), fill_value=255, dtype=np.uint8)

    # Generate some monotonically increasing timestamp
    timestamp = i * 1000

    # Encapsulate the image in a frame that has a timestamp, this does not copy the pixel data
    frame = dv.Frame(timestamp, image)

    # Write the frame
    writer.writeFrame(frame)
